cpp/ql/src/experimental/Security/CWE/CWE-120/MemoryUnsafeFunctionScan.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Scanf function without a specified length
+ * @description Use of one of the scanf functions without a specified length.
+ * @kind problem
+ * @problem.severity warning
+ * @id cpp/memory-unsafe-function-scan
+ * @tags reliability
+ *       security
+ *       external/cwe/cwe-120
+ */
+
+import cpp
+import semmle.code.cpp.commons.Scanf
+
+from FunctionCall call, ScanfFunction sff
+where
+  call.getTarget() = sff and
+  (
+    call.getArgument(sff.getFormatParameterIndex()).toString().regexpMatch(".*%s.*")
+    or
+    call.getArgument(sff.getFormatParameterIndex()).toString() = (".*%ls.*")
```

Comment:
The second case should be `.regexpMatch` as well.

Otherwise LGTM.

Reply:
I thought queries should not rely on toString calls?
Why is this ok in this case?

Reply:
Good catch. Of course, both regexes can be merged into one as `.*%l?s.*`. That should simplify things.

Reply:
> I thought queries should not rely on toString calls?

Good point.  I think `getValue()` will work just as well for constant strings, and is safer / more predictable / may involve less computation.

Reply:
Thanks for all of the feedback. I now use getValue() with a single `.regexmatch` of `.*%l?s.*`